svd-rs/src/array.rs

```diff
@@ -0,0 +1,122 @@
+use super::{DimElement, Name};
+use core::ops::{Deref, DerefMut};
+use std::borrow::Cow;
+
+/// A single SVD instance or array of instances
+#[derive(Clone, Debug, PartialEq)]
+pub enum SingleArray<T> {
+    /// A single instance
+    Single(T),
+    /// An array of instances
+    Array(T, DimElement),
+}
+
+impl<T> Deref for SingleArray<T> {
+    type Target = T;
+
+    fn deref(&self) -> &T {
+        match self {
+            Self::Single(info) => info,
+            Self::Array(info, _) => info,
+        }
+    }
+}
+
+impl<T> DerefMut for SingleArray<T> {
+    fn deref_mut(&mut self) -> &mut T {
+        match self {
+            Self::Single(info) => info,
+            Self::Array(info, _) => info,
+        }
+    }
+}
+
+impl<T> SingleArray<T> {
+    /// Return `true` if instance is single
+    pub const fn is_single(&self) -> bool {
+        matches!(self, Self::Single(_))
+    }
+    /// Return `true` if it is an array
+    pub const fn is_array(&self) -> bool {
+        matches!(self, Self::Array(_, _))
+    }
+}
+
+impl<T> Name for SingleArray<T>
+where
+    T: Name,
+{
+    fn name(&self) -> &str {
+        T::name(self)
+    }
+}
+
+impl<T> SingleArray<T>
+where
+    T: Name,
+{
+    /// Returns list of single or array names
+    pub fn names(&self) -> Vec<Cow<str>> {
+        match self {
+            Self::Single(info) => vec![info.name().into()],
+            Self::Array(info, dim) => dim
+                .indexes()
+                .map(|i| info.name().replace("[%s]", &i).replace("%s", &i).into())
```

Comment:
Maybe this should be a generic Fn to plug in a choosen function to map into Cow<str> ?

Reply:
Sorry, but I don't understand what you means.

Reply:
@Emilgardis I've deleted additional array functions for now. I'll make another PR for this